tool/gravity/cli/config.go

```diff
@@ -349,6 +349,17 @@ func (i *InstallConfig) CheckAndSetDefaults(validator resources.Validator) (err
 	if err := checkLocalAddr(i.AdvertiseAddr); err != nil {
 		return trace.Wrap(err)
 	}
+
+	if err := validate.NetworkOverlap(i.AdvertiseAddr, i.ServiceCIDR, fmt.Sprintf("The selected advertise address %v conflicts with the service network CIDR range %v. Please specify a different service CIDR via --service-cidr flag.",
```

Comment:
Like I hinted at previously, this will miss the service/pod CIDR configuration coming from a resource file. I'd look into [validateResources](https://github.com/gravitational/gravity/blob/version/7.0.x/tool/gravity/cli/config.go#L650) instead.